docs(secrets): rework secrets section (#869)
kumahq/kuma-website: docs/docs/dev/security/secrets.md

# Secrets

> The `Secret` resource implements the functionality of storing sensitive information.

Reviewer: ```suggestion
The `Secret` resource enables users to store sensitive data.
```
